rawfile/utils/snapshot_manager.py

```diff
@@ -0,0 +1,156 @@
+from pathlib import Path
+import time
+from dataclasses import dataclass
+import consts
+from utils.commands import run
+
+from utils.errors import SnapshotCreateVolumeInUse
+from utils.lock import VolLock
+from utils.rawfile import attached_loops, img_file, snapshots_dir
+from glob import glob
+from filesystem import from_device as fs_from_device
+
+
+@dataclass
+class Snapshot:
+    name: str
+    volume_id: str
+    size_bytes: int
+    creation_time: float
+    snapshot_id: str
+    ready: bool
+
+
+@dataclass
+class SnapshotList:
+    data: list[Snapshot]
+    next_token: int | None
+
+
+class SnapshotManager:
+    def _get_snapshot_path(self, volume_id: str, name: str) -> Path:
+        return snapshots_dir(volume_id) / f"{name}.img"
+
+    def create_snapshot(
```

Comment:
in case of pod/node restart do we always retry the tasks ahead of any csi operations? or is there a chance a csi operation could get through?

Reply:
There is a chance for that, we will just mark them as failed and retry worker will pick them

Reply:
We can make it unfreeze all on startup

or we can say that this is expected while snapshot creation is in progress

Reply:
If we don't unfreeze but accept the csi nodeStage call, we risk being frozen forever if not cow, because in that case snapshot task would fail with `SnapshotCreateVolumeInUse`?

Reply:
If we have enabled freeze if we not fail with SnapshotCreateVolumeInUse

Reply:
oh I see, I misread the code below
then the app would just get stuck until the snapshot is taken, I think this is fine then?

Reply:
Yep it is expected